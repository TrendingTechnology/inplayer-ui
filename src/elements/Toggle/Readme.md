Toggle is a input component for flipswitch. You can use it directly like this:

```jsx static
import { Toggle } from 'inplayer-ui';
import { Label } from 'inplayer-ui';

function Page(props) {
  return <Toggle id="toggle1" />
<Label htmlFor="toggle1">Checkbox1</Label>
}
```

### Demo

```jsx
<Toggle id="toggle1" />
<Label htmlFor="toggle1">Checkbox1</Label>
```
