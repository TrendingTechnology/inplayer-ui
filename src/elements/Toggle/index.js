import styled from 'styled-components';
import { uiColors } from 'utils';

const Toggle = styled.input`
  width: 34px;
  background: ${uiColors('lightGray')};
  border: 1px solid ${uiColors('gray')};
  border-radius: 50px;
  padding: 9px;
  margin: 12px 0;
  transition: all ease 500ms;
  overflow: visible;

  &:checked {
    border: 1px solid ${uiColors('accentBlue')};
    background: ${uiColors('accentBlue')};
  }

  &:hover,
  &:checked:hover {
    background: ${uiColors('accentBlueFade')};
    border: 1px solid ${uiColors('accentBlue')};
  }

  &::after {
    content: '';
    display: block;
    position: absolute;
    border: 1px solid ${uiColors('gray')};
    top: -2px;
    left: -2px;
    width: 20px;
    height: 20px;
    z-index: 2;
    background: #fff;
    border-radius: 50px;
    transition: all ease 300ms;
  }

  &:hover::after {
    border: 1px solid ${uiColors('accentBlue')};
  }

  &:checked::after {
    left: 13px;
  }
`;

/** @component */
export default Toggle;
