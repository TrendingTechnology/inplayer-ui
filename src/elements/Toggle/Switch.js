// @flow
import React from 'react';
import styled from 'styled-components';

import colors from 'config/colors';
import Toggle from 'react-toggle';

type Props = {
  checked: boolean,
  onChange: Function,
  disabled: boolean,
};

const ToggleContainer = styled.span`
  margin-right: 8px;

  .react-toggle {
    touch-action: pan-x;
    display: inline-block;
    position: relative;
    cursor: pointer;
    background-color: transparent;
    border: 0;
    padding: 0;
    user-select: none;
  }

  .react-toggle-screenreader-only {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }

  .react-toggle-thumb {
    transition: all 0.5s cubic-bezier(0.23, 1, 0.32, 1) 0ms;
    position: absolute;
    top: -1px;
    left: 0;
    width: 22px;
    height: 22px;
    border-radius: 50%;
    border: 1px solid ${colors.gray};
    background: ${colors.white};
    box-sizing: border-box;
  }

  .react-toggle-track {
    width: 34px;
    height: 18px;
    padding: 0;
    border-radius: 30px;
    background-color: ${colors.lightGray};
    transition: all 0.2s ease;
    border: 1px solid ${colors.gray};
  }

  .react-toggle--disabled {
    background: #00000005 !important;
    cursor: auto;
  }

  .react-toggle--disabled .react-toggle-track {
    background: #00000005 !important;
    border: 1px solid #d7dde5 !important;
  }

  .react-toggle--disabled .react-toggle-thumb {
    background: #f0f3f6;
    border: 1px solid #d7dde5;
  }

  .react-toggle--checked .react-toggle-track {
    border: 1px solid ${colors.accentBlue};
    background: ${colors.accentBlue};
  }

  .react-toggle--checked .react-toggle-thumb {
    left: 14px;
  }

  .react-toggle:hover:not(.react-toggle--disabled) .react-toggle-track {
    background: ${colors.accentBlueFade};
    border: 1px solid ${colors.accentBlue};
  }

  .react-toggle--checked:hover:not(.react-toggle--disabled) .react-toggle-track {
    background: rgba(0, 170, 230, 0.06);
    border: 1px solid ${colors.accentBlue};
  }

  .react-toggle:hover:not(.react-toggle--disabled) .react-toggle-thumb {
    border: 1px solid ${colors.accentBlue};
  }

  .react-toggle-track-check {
    position: absolute;
    width: 14px;
    height: 10px;
    top: 0;
    bottom: 0;
    margin-top: auto;
    margin-bottom: auto;
    line-height: 0;
    left: 8px;
    opacity: 0;
    transition: opacity 0.25s ease;
  }

  .react-toggle--checked .react-toggle-track-check {
    opacity: 1;
    transition: opacity 0.25s ease;
  }

  .react-toggle-track-x {
    position: absolute;
    width: 10px;
    height: 10px;
    top: 0;
    bottom: 0;
    margin-top: auto;
    margin-bottom: auto;
    line-height: 0;
    right: 10px;
    opacity: 1;
    transition: opacity 0.25s ease;
  }

  .react-toggle--checked .react-toggle-track-x {
    opacity: 0;
  }
`;

const Switcher = ({ checked, onChange, disabled }: Props) => (
  <ToggleContainer>
    <Toggle disabled={disabled} defaultChecked={checked} onChange={onChange} icons={false} />
  </ToggleContainer>
);

/** @component */
export default Switcher;
