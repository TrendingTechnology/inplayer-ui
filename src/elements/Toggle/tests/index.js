import { generateSnapshots } from 'helpers';

import Toggle from '../index';

generateSnapshots(Toggle);
