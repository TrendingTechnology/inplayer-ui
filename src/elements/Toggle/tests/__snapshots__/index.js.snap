// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #d7dde5;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #a0a3aa;
  font-weight: 600;
  border: 1px solid #d7dde5;
  padding: 10px 20px 11px;
  border-radius: 3px;
  height: 1.125rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  width: 1.125rem;
  cursor: pointer;
  vertical-align: middle;
}

.c0:hover,
.c0:focus {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c0:hover span,
.c0:focus span {
  color: #00aae6;
}

<button
  className="c0"
/>
`;

exports[`Toggle renders correctly 1`] = `
.c0 {
  width: 34px;
  background: #f0f3f6;
  border: 1px solid #d7dde5;
  border-radius: 50px;
  padding: 9px;
  margin: 12px 0;
  -webkit-transition: all ease 500ms;
  transition: all ease 500ms;
  overflow: visible;
}

.c0:checked {
  border: 1px solid #00aae6;
  background: #00aae6;
}

.c0:hover,
.c0:checked:hover {
  background: #00aae60f;
  border: 1px solid #00aae6;
}

.c0::after {
  content: '';
  display: block;
  position: absolute;
  border: 1px solid #d7dde5;
  top: -2px;
  left: -2px;
  width: 20px;
  height: 20px;
  z-index: 2;
  background: #fff;
  border-radius: 50px;
  -webkit-transition: all ease 300ms;
  transition: all ease 300ms;
}

.c0:hover::after {
  border: 1px solid #00aae6;
}

.c0:checked::after {
  left: 13px;
}

<input
  className="c0"
/>
`;
